tracker/event_tracker.go

```diff
@@ -79,30 +87,15 @@ func (e *EventTracker) Start(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
-
-	go func() {
-		if err := blockTracker.Init(); err != nil {
-			e.logger.Error("failed to init blocktracker", "error", err)
-
-			return
-		}
-
-		if err := blockTracker.Start(); err != nil {
-			e.logger.Error("failed to start blocktracker", "error", err)
-		}
-	}()
+	if err := tt.Sync(ctx); err != nil {
```

Comment:
I'm just a bit puzzled is it ok to invoke `Sync` synchronously?

Reply:
Yes this is a good point.

I am not able to test a situation where `func (t *Tracker) findAncestor(block, pivot *ethgo.Block) (uint64, error)` might involve a large range of blocks and many RPC requests.

My main question is - should the runtime be synchronous here notwithstanding the potential in delay given the importance of the event tracker for a running validator?

If we are not happy being synchronous here in the worst case, should the client exit on failure of `tt.Sync()`?

Reply:
> My main question is - should the runtime be synchronous here notwithstanding the potential in delay given the importance of the event tracker for a running validator?

IMO we could live with potential delays due to the async nature of the tracker synchronization.

> If we are not happy being synchronous here in the worst case, should the client exit on failure of tt.Sync() despite being async?

Not sure about this either. For example, for the `StateSyncRelayer` component, which executes bridge events (transfers assets from the root chain to the child chain) I don't feel it is critical enough to shut down the validator in case of `Sync` failure. On the other hand for the `stateSyncManager,` it might make sense.

I've talked a bit with @goran-ethernal and his idea was maybe to consider propagating errors using some error channel. In that case, we could once again closely analyze whether it makes sense to revert everything to async behavior or do it only partially (e.g. make only `Sync` invocation async).

Reply:
> propagating errors using some error channel. In that case, we could once again closely analyze whether it makes sense to revert everything to async behavior or do it only partially (e.g. make only Sync invocation async).

Makes sense. I will update the PR